core/src/main/scala/org/apache/spark/ui/jobs/AllJobsPage.scala

```diff
@@ -295,6 +296,12 @@ private[ui] class AllJobsPage(parent: JobsTab) extends WebUIPage("") {
                 <strong>Total Uptime: </strong>
                 {UIUtils.formatDuration(System.currentTimeMillis() - startTime)}
               </li>
+            } else {
```

Comment:
As @srowen pointed out, you can reduce these branches above. Also, we should consider incomplete apps when we use the history server.